Algorithms/Sorters/CycleSorter.cs

```diff
@@ -1,14 +1,31 @@
-namespace cycle_sort
+﻿using System.Collections.Generic;
+
+namespace Algorithms.Sorters
 {
-    internal class Program
+    /// <summary>
+    /// Cycle sort is an in-place, unstable sorting algorithm,
+    /// a comparison sort that is theoretically optimal in terms of the total
+    /// number of writes to the original array.
+    /// It is based on the idea that the permutation to be sorted can be factored
+    /// into cycles, which can individually be rotated to give a sorted result.
+    /// </summary>
+    public class CycleSorter : ISorter<int>
```

Comment:
Make class definition generic and use comparer to compare values because it's comparison sort.